src/lib/MovingAverageInterface.java

```diff
@@ -0,0 +1,23 @@
+package lib;
+
+
+/**
+ * 
+ * @author khenr A moving average is used to keep track of a windowed average that has values
+ *         constantly being added and removed.
+ */
+public interface MovingAverageInterface {
+
+  /**
+   * Removes the first value from the average ands a new one to the average.
```

Comment:
Make specific that value is only removed if more than windowSize values have been added